File: app/src/main/java/com/example/mesplats/AppDatabase.java
package com.example.mesplats;

import android.content.Context;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;


@Database(
        entities=[Ingredient::class, Recipe::class],
        version = 1)
abstract class AppDatabase : RoomDatabase() {

abstract fun getIngredientDAO(): IngredientDAO

//        companion object {
//private var INSTANCE: AppDatabase? = null
//
//        fun getAppDatabase(context: Context): AppDatabase {
//        if (INSTANCE == null) {
//synchronized(AppDatabase::class) {
//        INSTANCE = Room
//        .databaseBuilder(context.applicationContext, AppDatabase::class.java, "db")
//        .allowMainThreadQueries()
//        .fallbackToDestructiveMigration()
//        .build()
//        }
//        }
//        return INSTANCE as AppDatabase
//        }
//        }
        }

File: app/src/main/java/com/example/mesplats/AppDatabase.kt
package com.example.mesplats

import androidx.fragment.app.FragmentActivity
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase


@Database(
    entities=[Ingredient::class, Recipe::class, RecipeIngredient::class],
    version = 3
)
abstract class AppDatabase : RoomDatabase() {

    abstract fun getIngredientDao(): IngredientDao
    abstract fun getRecipeDao(): RecipeDao
    abstract fun getRecipeIngredientDao(): RecipeIngredientDao

    companion object {
        private var INSTANCE: AppDatabase? = null

        fun getAppDatabase(context: FragmentActivity?): AppDatabase {
            if (INSTANCE == null) {
                synchronized(AppDatabase::class) {
                    if (context != null) {
                        INSTANCE = Room
                            .databaseBuilder(context.applicationContext, AppDatabase::class.java, "db")
                            .allowMainThreadQueries()
                            .fallbackToDestructiveMigration()
                            .build()
                    }
                }
            }
            return INSTANCE as AppDatabase
        }
    }
}
